test/integration/authorization_test.go

```diff
@@ -78,6 +78,44 @@ func TestRestrictedAccessForProjectAdmins(t *testing.T) {
 	// }
 }
 
+func TestOnlyResolveRolesForBindingsThatMatter(t *testing.T) {
+	startConfig, err := StartTestMaster()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	openshiftClient, _, err := startConfig.GetOpenshiftClient()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	addValerie := &policy.AddUserOptions{
+		RoleNamespace:    "master",
+		RoleName:         "view",
+		BindingNamespace: "master",
+		Client:           openshiftClient,
+		Users:            []string{"anypassword:valerie"},
+	}
+	if err := addValerie.Run(); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if err = openshiftClient.Roles("master").Delete("view"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
```

Comment:
maybe try to run addValerie again and make sure you get an error when the role is invalid?